sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateEncoder.scala

```diff
@@ -147,35 +147,55 @@ class PrefixKeyScanStateEncoder(
   // Reusable objects
   private val joinedRowOnKey = new JoinedRow()
 
-  override def encodeKey(row: UnsafeRow): Array[Byte] = {
+  override def encodeKey(row: UnsafeRow, vcfId: Option[Short]): Array[Byte] = {
+    val hasVirtualColFamilyPrefix: Boolean = vcfId.isDefined
     val prefixKeyEncoded = encodeUnsafeRow(extractPrefixKey(row))
     val remainingEncoded = encodeUnsafeRow(remainingKeyProjection(row))
+    val offSetForColFamilyPrefix =
```

Comment:
This function seems repeated right ? can we just put it in the base class maybe ?

Reply:
Move it out to the base trait.